test/e2e/throughputanomalydetection_test.go

```diff
@@ -0,0 +1,421 @@
+			"1.000": "true",
+			"3.973": "true"},
+		"EWMA": {
+			"2.700": "true",
+			"1.550": "true",
+			"9.755": "true"},
+		"DBSCAN": {
+			"1000": "true",
+			"1005": "true",
+			"5000": "true",
+			"3260": "true",
+			"2058": "true"},
+	}
+	for _, algo := range algoNames {
+		_, jobName, err := tadrunJob(t, data, algo)
+		require.NoError(t, err)
+		err = data.podWaitForReady(defaultTimeout, jobName+"-driver", flowVisibilityNamespace)
+		require.NoError(t, err)
+		err = waitTADJobComplete(t, data, jobName, tadjobCompleteTimeout)
+		require.NoErrorf(t, err, "Throughput Anomaly Detection Spark job failed to complete")
+		stdout, err := tadretrieveJobResult(t, data, jobName)
+		resultArray := strings.Split(stdout, "\n")
+		assert := assert.New(t)
+		length := len(resultArray)
+		assert.GreaterOrEqualf(length, 3, "stdout: %s", stdout)
+		assert.Containsf(stdout, "Throughputs", "stdout: %s", stdout)
+		assert.Containsf(stdout, "algoCalc", "stdout: %s", stdout)
+		assert.Containsf(stdout, "anomaly", "stdout: %s", stdout)
+
+		for i := 1; i < length; i++ {
+			// check metrics' value
+			resultArray[i] = strings.TrimSpace(resultArray[i])
+			if resultArray[i] != "" {
+				resultArray[i] = strings.ReplaceAll(resultArray[i], "\t", " ")
+				tadoutputArray := strings.Fields(resultArray[i])
+				anomaly_output := tadoutputArray[10]
+				assert.Equal(11, len(tadoutputArray), "tadoutputArray: %s", tadoutputArray)
+				switch algo {
+				case "ARIMA":
+					algo_throughput := tadoutputArray[9][:5]
+					assert.Equal(anomaly_output, result_map["ARIMA"][algo_throughput], "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
+				case "EWMA":
+					algo_throughput := tadoutputArray[9][:5]
+					assert.Equal(anomaly_output, result_map["EWMA"][algo_throughput], "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
+				case "DBSCAN":
+					throughput := tadoutputArray[8][:4]
+					assert.Equal(anomaly_output, result_map["DBSCAN"][throughput], "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
+				}
+			}
+		}
+		require.NoError(t, err)
+		_, err = taddeleteJob(t, data, jobName)
+		require.NoError(t, err)
+	}
+}
+
+// waitJobComplete waits for the anomaly detection Spark job completes
+func waitTADJobComplete(t *testing.T, data *TestData, jobName string, timeout time.Duration) error {
+	stdout := ""
+	err := wait.PollImmediate(defaultInterval, timeout, func() (bool, error) {
+		stdout, err := tadgetJobStatus(t, data, jobName)
+		require.NoError(t, err)
+		if strings.Contains(stdout, "Status of this anomaly detection job is COMPLETED") {
+			return true, nil
+		}
+		// Keep trying
+		return false, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf("anomaly detection Spark job not completed after %v\nstatus:%s", timeout, stdout)
+	} else if err != nil {
+		return err
+	}
+	return nil
+}
+
+func tadrunJob(t *testing.T, data *TestData, algotype string) (stdout string, jobName string, err error) {
```

Comment:
We have this series of functions `runJob`, `getJobStatus`, `listJobs`, `deleteJob`, `retrieveJobResult` in the policy recommendation e2e test, the only difference here is we are passing different command string. Could you please reuse these functions for these two e2e tests?

Reply:
added them to util.go